### DISCUSSION TITLE
请问milvus单个cluster支持10m个partition的提升功能何时发布？会在2.4吗？谢谢！

### DISCUSSION THREAD

**Discussion by xiaobingxia-at** [milvus-io/milvus#38685] [Q&A and General discussion]
另外，改善load_collection/release_connection的功能何时发布啊？是在2.4还是2.5？谢谢！

**Comment by yhmo** [comment]
10 million partitions? 这恐怕很难，可以作为长期目标。
目前目标先是能到1万个，顺利的话2.5的后期版本也许会有。

**Reply by xiaobingxia-at** [reply]
@xiaofan-luan 您上次说，可以支持10m个partition？谢谢！

**Reply by xiaobingxia-at** [reply]
另外，改善load_collection/release_connection速度的功能何时发布啊？是在2.4还是2.5？谢谢！

**Comment by xiaofan-luan** [comment]
目前实验室环境支持了5K collection，每个collection 1000 partition

@bigsheeper  正在进行更多的实验，预计下个月就可以发布

**Reply by xiaobingxia-at** [reply]
感谢！！！

**Reply by xiaobingxia-at** [reply]
@xiaofan-luan @bigsheeper 能否请教一下大概得优化方向，是提升coordinator的并行度所以它可以处理更多的segments，还是尽量减少了针对segment的，在coordinator层面的工作量。
目前我针对大量segments做的优化是，尽量降低了mix/l0 compaction，balancing，gc的频率，变成几小时一次。希望不要让coordinator过度负载。

**Reply by bigsheeper** [reply]
可以看看这个 issue 关联的 PR https://github.com/milvus-io/milvus/issues/37630
主要方向是优化全局锁，优化 meta 结构等，此外还优化了内存占用、重启恢复速度等，相关 PR 已经在审核流程中
